### PR TITLE
rafthttp: add Transport.Cut/MendPeer

### PR DESCRIPTION
**Pull request by gyuho**
From https://github.com/coreos/etcd/pull/6140.

/cc @xiang90 @heyitsanthony 

Thanks.

### CODE REVIEW

File: rafthttp/transport.go [hunk 1]
```diff
@@ -206,6 +206,36 @@ func (t *Transport) Stop() {
 	t.remotes = nil
 }
 
+// PausePeer drops messages to the specified peer.
+func (t *Transport) PausePeer(id types.ID) {
```
Comment (xiang90) [inline rafthttp/transport.go]: BlockPeer?

File: rafthttp/transport.go [hunk 2]
```diff
@@ -206,6 +206,36 @@ func (t *Transport) Stop() {
...
+	if pok {
+		p.(Pausable).Pause()
+	}
+	if gok {
+		g.Pause()
+	}
+}
+
+// ResumePeer recovers the send pause to the given peer.
+func (t *Transport) ResumePeer(id types.ID) {
```
Comment (xiang90) [inline rafthttp/transport.go]: UnblockPeer?

**Comment by gyuho** [comment]
@xiang90 Fixed. PTAL. Thanks.

File: rafthttp/transport.go [hunk 3]
```diff
@@ -206,6 +206,36 @@ func (t *Transport) Stop() {
...
+
+	if pok {
+		p.(Pausable).Pause()
+	}
+	if gok {
+		g.Pause()
+	}
+}
+
+// UnblockPeer recovers the send pause to the given peer.
```
Comment (xiang90) [inline rafthttp/transport.go]: UnblockPeer recovers the message dropping behavior of the given peer.

**Comment by xiang90** [comment]
lgtm. defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
I don't think it should be BlockPeer/UnblockPeer; that sounds like it'll queue up / wait when really it's dropping packets. What's wrong with PausePeer/ResumePeer? /cc @xiang90

**Comment by xiang90** [comment]
@heyitsanthony There is already a Pause/Resume there, which pause the local member's entire transport. PausePeer sounds like we can pause remote peer's transport. But actually we only pause the local member's transport to that peer. If we want to make them more consistent, we can do PauseAll (pause all) and PausePeer? Then the targets are clearly not local member in both cases.

**Comment by heyitsanthony** [comment]
@xiang90 but if `PausePeer` paused the remote peer's transport so that it drops all messages like `Pause`, how would `ResumePeer` make sense? I think it's OK...

**Comment by heyitsanthony** [comment]
Maybe `CutPeer`/`MendPeer`?

**Comment by xiang90** [comment]
sounds good.

**Pull request merged**